src/libs/antares/study/parts/hydro/hydroreservoirlevels.cpp

```diff
@@ -0,0 +1,217 @@
+
+#include <yuni/yuni.h>
+#include <yuni/io/file.h>
+
+#include <antares/exception/LoadingError.hpp>
+#include <antares/inifile/inifile.h>
+#include <antares/logs/logs.h>
+#include <antares/study/parts/hydro/hydroreservoirlevels.h>
+#include "antares/study/study.h"
+
+namespace fs = std::filesystem;
+
+namespace Antares::Data
+{
+
+static std::unique_ptr<ReservoirLevelsLoader> createReservoirLevelLoader(
+  Parameters::Compatibility::HydroRuleCurves hydroRuleCurves,
+  const std::filesystem::path& filePath,
+  const std::string& areaID,
+  Matrix<double>& standardReservoirLevelMatrix,
+  TimeSeries& max,
+  TimeSeries& avg,
+  TimeSeries& min)
+{
+    switch (hydroRuleCurves)
+    {
+    case Parameters::Compatibility::HydroRuleCurves::Single:
+    {
+        return std::make_unique<StandardReservoirLevelsLoader>(filePath,
+                                                               areaID,
+                                                               standardReservoirLevelMatrix,
+                                                               max,
+                                                               avg,
+                                                               min);
+    }
+    case Parameters::Compatibility::HydroRuleCurves::Scenarized:
+    {
+        return std::make_unique<ScenarizedReservoirLevelLoader>(filePath, areaID, max, avg, min);
+    }
+    default:
+        throw std::invalid_argument("Value not supported for hydro rule curves compatibility");
+    }
+}
+
+ReservoirLevels::ReservoirLevels(TimeSeriesNumbers& timeseriesNumbers):
+    max(timeseriesNumbers),
+    min(timeseriesNumbers),
+    avg(timeseriesNumbers),
+    timeseriesNumbers(timeseriesNumbers)
+{
+    timeseriesNumbers.registerSeries(&max, "max-reservoir-level");
+    timeseriesNumbers.registerSeries(&min, "min-reservoir-level");
+    timeseriesNumbers.registerSeries(&avg, "avg-reservoir-level");
+
+    max.reset(1L, DAYS_PER_YEAR);
+    max.fill(1.0);
+    avg.reset(1L, DAYS_PER_YEAR);
+    avg.fill(0.5);
+    min.reset(1L, DAYS_PER_YEAR);
+    standardReservoirLevelMatrix.reset(3L, DAYS_PER_YEAR, true);
+    standardReservoirLevelMatrix.fillColumn(ReservoirLevels::maximum, 1.);
+    standardReservoirLevelMatrix.fillColumn(ReservoirLevels::average, 0.5);
+}
+
+bool ReservoirLevels::forceReload(bool reload) const
+{
+    bool ret = true;
+    ret = max.forceReload(reload) && ret;
+    ret = min.forceReload(reload) && ret;
+    ret = avg.forceReload(reload) && ret;
+    ret = standardReservoirLevelMatrix.forceReload(reload) && ret;
+
+    return ret;
+}
+
+void ReservoirLevels::markAsModified() const
+{
+    max.markAsModified();
+    min.markAsModified();
+    avg.markAsModified();
+    standardReservoirLevelMatrix.markAsModified();
+}
+
+bool ReservoirLevels::loadReservoirLevels(
+  const std::string& areaID,
+  const std::filesystem::path& folder,
+  bool usedBySolver,
+  Parameters::Compatibility::HydroRuleCurves hydroRuleCurves)
```

Comment:
Isn't there a `ReservoirLevelsLoader` class ? If so, why does `ReservoirLevels` still has some code related to loading ?